### PR TITLE
use build only hash in build reporting on build only

### PR DESCRIPTION
**Pull request by ewlu**
fixes the baseline hash reporting as seen in https://github.com/ewlu/gcc-precommit-ci/issues/449#issuecomment-1776520951

### CODE REVIEW

File: .github/workflows/test-regression.yaml [hunk 1]
```diff
@@ -865,7 +901,11 @@ jobs:
       - name: Build test starting report
         if: ${{ inputs.build_only != 'true' }}
```
Comment (patrick-rivos) [inline .github/workflows/test-regression.yaml:902]: We already have an `if build_only == true` here. Can't we rely on this for filtering?

File: .github/workflows/test-regression.yaml [hunk 2]
```diff
@@ -948,7 +988,11 @@ jobs:
       - name: Build test complete report
         if: ${{ inputs.build_only != 'true' }}
```
Comment (patrick-rivos) [inline .github/workflows/test-regression.yaml:989]: Same here

**Review by patrick-rivos — CHANGES_REQUESTED** [review]
Also needs changes here: https://github.com/ewlu/gcc-precommit-ci/blob/cea3a1bb6f90b215ec3a124dbcad04953d385685/.github/workflows/generate-precommit-summary.yaml#L176

**Comment by ewlu** [comment]
> Also needs changes here:
> 
> https://github.com/ewlu/gcc-precommit-ci/blob/cea3a1bb6f90b215ec3a124dbcad04953d385685/.github/workflows/generate-precommit-summary.yaml#L176

We don't need to worry about this one since it doesn't produce a comment report. the `-failure` flag only searches for build failures

File: .github/workflows/test-regression.yaml (hunk 1 above)
Comment (ewlu) [reply .github/workflows/test-regression.yaml]: Missed that. Also forgot that the test reports on the non-self-hosted runners also have the if condition

**Review by patrick-rivos — APPROVED** [review]

**Pull request merged**